Automattic/simplenote-ios
# Show "Trash" button when editing Tags from the Sidebar

### Fix
In this PR we're adding a trash button to the tag cells when in edit mode. Trash button is used instead of a standard UITableView deletion mechanism.

* `SPTagListViewCell` is converted to swift and renamed to `TagListViewCell`
* `TagListViewCell` is used only for tags, for other rows we now use `Value1TableViewCell `
* Before deleting a tag we now show a confirmation message

Closes #1042 

**Before**
|||
|-|-|
|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 14 55 19](https://user-images.githubusercontent.com/54751/104461284-7249a980-55af-11eb-99bd-6d805397bae6.png)|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 14 55 17](https://user-images.githubusercontent.com/54751/104461292-75449a00-55af-11eb-91c3-128274233ab9.png)|

**After**
|||
|-|-|
|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 15 55 53](https://user-images.githubusercontent.com/54751/104468609-eee08600-55b7-11eb-918d-5c679ae3d67d.png)|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 16 06 43](https://user-images.githubusercontent.com/54751/104469941-619e3100-55b9-11eb-8197-966c634a2911.png)|
|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 15 55 55](https://user-images.githubusercontent.com/54751/104468620-f142e000-55b7-11eb-9e23-3cfb0e480de5.png)|![Simulator Screen Shot - iPhone 12 - 2021-01-13 at 16 06 41](https://user-images.githubusercontent.com/54751/104469953-64992180-55b9-11eb-9371-5860559d712f.png)|


### Test
1. Open sidebar
2. Enter edit mode

- [x] Check that Trash button is shown next to each tag
- [x] Tap on Trash button, confirmation is show
- [x] Cancelling the confirmation doesn't delete a tag
- [x] Tapping "Delete Tag" in the confirmation deletes a tag
- [x] Long press on a tag and pick Delete. Check that confirmation is shown.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in https://github.com/Automattic/simplenote-ios/commit/78e73be72b23b86467db793352b37e9398b036e6 with:

> Updated tags editing experience #1042
